File: PriceComparisonMonitoring/backend/utils/response.go
package utils

import (
	"net/http"

	"github.com/gin-gonic/gin"
)

type Response struct {
	Code    int         `json:"code"`
	Message string      `json:"message"`
	Data    interface{} `json:"data,omitempty"`
}

type PageResult struct {
	List     interface{} `json:"list"`
	Total    int64       `json:"total"`
	Page     int         `json:"page"`
	PageSize int         `json:"page_size"`
}

func SuccessResponse(data interface{}) Response {
	return Response{
		Code:    http.StatusOK,
		Message: "success",
		Data:    data,
	}
}

func ErrorResponse(code int, message string) Response {
	return Response{
		Code:    code,
		Message: message,
	}
}

func PageResponse(list interface{}, total int64, page, pageSize int) Response {
	return Response{
		Code:    http.StatusOK,
		Message: "success",
		Data: PageResult{
			List:     list,
			Total:    total,
			Page:     page,
			PageSize: pageSize,
		},
	}
}

func SendSuccess(c *gin.Context, data interface{}) {
	c.JSON(http.StatusOK, SuccessResponse(data))
}

func SendError(c *gin.Context, code int, message string) {
	c.JSON(code, ErrorResponse(code, message))
}

func SendPage(c *gin.Context, list interface{}, total int64, page, pageSize int) {
	c.JSON(http.StatusOK, PageResponse(list, total, page, pageSize))
}

File: APIMockContractTestingPlatform/src/utils/response.js
function success(res, data, message = 'success', statusCode = 200) {
  res.status(statusCode).json({
    status: 'success',
    message,
    data
  });
}

function paginate(res, data, total, page, pageSize, message = 'success') {
  res.status(200).json({
    status: 'success',
    message,
    data,
    pagination: {
      total,
      page,
      pageSize,
      totalPages: Math.ceil(total / pageSize)
    }
  });
}

module.exports = {
  success,
  paginate
};
